improver/utilities/spatial.py

```diff
@@ -538,9 +538,31 @@ def lat_lon_determine(cube: Cube) -> Optional[CRS]:
     return trg_crs
 
 
-def transform_grid_to_lat_lon(cube: Cube) -> Tuple[ndarray, ndarray]:
+def get_grid_y_x_values(cube: Cube) -> Tuple[ndarray, ndarray]:
+    """Extract the y and x coordinate values of each points in the cube.
+
+    Args:
+        cube:
+            Cube with points to extract
+    Returns:
+        - Array of shape cube.data.shape containing y coordinate values
+        - Array of shape cube.data.shape containing x coordinate values
     """
-    Calculate the latitudes and longitudes of each points in the cube.
+    x_points = cube.coord(axis="x").points
+    y_points = cube.coord(axis="y").points
+
+    x_zeros = np.zeros_like(x_points)
+    y_zeros = np.zeros_like(y_points)
+
+    # Broadcast x points and y points onto grid
+    all_x_points = y_zeros.reshape(len(y_zeros), 1) + x_points
+    all_y_points = y_points.reshape(len(y_points), 1) + x_zeros
```

Comment:
I reviewed this before seeing that you had refactored an existing bit of code!

I'm not convinced this will work if your grid is square. Try something like this instead.
```
result = []
for axis in "yx":
    coord = cube.coord(axis=axis)
    coord_dim, = cube.coord_dims(coord)
    source_shape = np.ones_like(cube.shape)
    source_shape[coord_dim] = cube.shape[coord_dim]
    broadcasted_array = np.broadcast_to(coord.points.reshape(source_shape), cube.shape)
    result.append(broadcasted_array)
return result
```

Reply:
Yes, I'm happy with updating the doc-string instead of the code.